src/mongo/db/query/datetime/date_time_support.cpp

```diff
@@ -122,6 +131,16 @@ TimeZone TimeZoneDatabase::utcZone() {
     return TimeZone{nullptr};
 }
 
+Date_t TimeZoneDatabase::fromString(StringData dateString) const {
+	timelib_time *t;
+	timelib_error_container *errors;
+
+	t = timelib_strtotime((char*) dateString.toString().c_str(), dateString.size(), &errors, _timeZoneDatabase, TimeZoneDatabase::getTZInfoWrapper);
```

Comment:
It looks like you're getting into some weird C++/C boundaries, where it's hard to generate a function pointer from a class method. This might help? https://stackoverflow.com/questions/1738313/c-using-class-method-as-a-function-pointer-type